gobblin-metrics/src/test/java/gobblin/metrics/KafkaReporterTest.java

```diff
@@ -0,0 +1,228 @@
+    Assert.assertEquals(host, metric.host);
+    Assert.assertEquals(env, metric.env);
+    Assert.assertEquals(2, metric.tags.size());
+    Assert.assertTrue(metric.tags.contains(tag1));
+    Assert.assertTrue(metric.tags.contains(tag2));
+
+  }
+
+  /**
+   * Expect a list of metrics with specific values.
+   * Fail if not all metrics are received, or some metric has the wrong value.
+   * @param it ConsumerIterator for Kafka topic
+   * @param expected map of expected metric names and their values
+   * @throws IOException
+   */
+  private void expectMetricsWithValues(ConsumerIterator<byte[], byte[]> it, Map<String, Double> expected)
+      throws IOException {
+    System.out.println("====Checking for expected messages and values. Will list all messages.====");
+    try {
+      while(it.hasNext()) {
+        KafkaReporter.Metric metric = nextMetric(it);
+        if (expected.containsKey(metric.name)) {
+          Assert.assertEquals(expected.get(metric.name), Double.parseDouble(metric.value.toString()));
+          expected.remove(metric.name);
+        }
+      }
+    } catch (ConsumerTimeoutException e) {
+      Assert.assertTrue(expected.isEmpty());
+    }
+  }
+
+  /**
+   * Expect a set of metric names. Will fail if not all of these metrics are received.
+   * @param it ConsumerIterator for Kafka topic
+   * @param expected set of expected metric names
+   * @param strict if set to true, will fail if receiving any metric that is not expected
+   * @throws IOException
+   */
+  private void expectMetrics(ConsumerIterator<byte[], byte[]> it, Set<String> expected, boolean strict)
+      throws IOException {
+    System.out.println("====Checking for expected messages. Will list all messages.====");
+    try {
+      while(it.hasNext()) {
+        KafkaReporter.Metric metric = nextMetric(it);
+        //System.out.println(String.format("expectedSet.add(\"%s\")", metric.name));
+        if (expected.contains(metric.name)) {
+          expected.remove(metric.name);
+        } else if (strict) {
+          Assert.assertTrue(false, "Message present in kafka not expected: " + metric.toString());
+        }
+      }
+    } catch(ConsumerTimeoutException e) {
+      Assert.assertTrue(expected.isEmpty());
+    }
+  }
+
+  /**
+   * Extract the next metric from the Kafka iterator
+   * Assumes existence of the metric has already been checked.
+   * @param it Kafka ConsumerIterator
+   * @return next metric in the stream
+   * @throws IOException
+   */
+  protected KafkaReporter.Metric nextMetric(ConsumerIterator<byte[], byte[]> it) throws IOException {
+    String nextMessage = new String(it.next().message());
+    KafkaReporter.Metric metric = mapper.readValue(nextMessage, KafkaReporter.Metric.class);
+    System.out.println(nextMessage);
+    return metric;
+  }
+
+  @AfterClass
+  public void after() {
+    try {
+      close();
+    } catch(Exception e) {
+      System.err.println("Failed to close Kafka server.");
```

Comment:
I suggest not printing anything in unit test. If there's an error due to bug, just let the test fail. If the test passes, nobody will read the printed messages.